cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/DefaultStore.java

```diff
@@ -466,37 +468,38 @@ public ApplicationMeta call(DatasetContext context) throws Exception {
   }
 
   @Override
-  public void addStream(final Id.Namespace id, final StreamSpecification streamSpec) {
+  public void addStream(final NamespaceId id, final StreamSpecification streamSpec) {
     txExecute(transactional, new TxRunnable() {
       @Override
       public void run(DatasetContext context) throws Exception {
-        getAppMetadataStore(context).writeStream(id.getId(), streamSpec);
+        getAppMetadataStore(context).writeStream(id.getNamespace(), streamSpec);
       }
     });
   }
 
   @Override
-  public StreamSpecification getStream(final Id.Namespace id, final String name) {
+  public StreamSpecification getStream(final NamespaceId id, final String name) {
     return txExecute(transactional, new TxCallable<StreamSpecification>() {
       @Override
       public StreamSpecification call(DatasetContext context) throws Exception {
-        return getAppMetadataStore(context).getStream(id.getId(), name);
+        return getAppMetadataStore(context).getStream(id.getNamespace(), name);
       }
     });
   }
 
   @Override
-  public Collection<StreamSpecification> getAllStreams(final Id.Namespace id) {
+  public Collection<StreamSpecification> getAllStreams(final NamespaceId id) {
     return txExecute(transactional, new TxCallable<Collection<StreamSpecification>>() {
       @Override
       public Collection<StreamSpecification> call(DatasetContext context) throws Exception {
-        return getAppMetadataStore(context).getAllStreams(id.getId());
+        return getAppMetadataStore(context).getAllStreams(id.getNamespace());
       }
     });
   }
 
   @Override
-  public FlowSpecification setFlowletInstances(final Id.Program id, final String flowletId, final int count) {
+  public FlowSpecification setFlowletInstances(final ProgramId programId, final String flowletId, final int count) {
+    final Id.Program id = programId.toId();
```

Comment:
why do you need this? should update any private methods in this class to use the new classes. 

Also, ``ProgramId`` is converted to ``Id.Program`` here, then converted back to ``ProgramId`` on line 512